teps/0031-tekton-bundles-cli.md

```diff
@@ -66,3 +71,83 @@ Provide a way for users to...
 * Users may publish to any repository they have access to.
 
 * Only valid Tekton Bundles are produced.
+
+## Proposal
+
+Amend the Tekton CLI to provide the following commands (more detail later):
+
+```shell
+# REF = OCI reference, eg. docker.io/myworkspace/mybundle:1.0
+# BUNDLE_OBJECT = path to a file or directory or STDIN of a valid Tekton Task or Pipeline object to be included in the
+# bundle.
+
+# Generates a Tekton Bundle from a set of 1+ bundle objects. Publishes the bundle to a remote repository using the
+# specified reference. Will properly split and parse input files that have multiple objects and rejects any input that
+# isn't a known and supported Tekton KIND.
+tkn bundle push <REF> [BUNDLE_OBJECT...]
```

Comment:
NIT: if people are used to `kubectl` they might like options like:

```
tkn bundle push <REF> -f [BUNDLE_OBJECT...]
tkn bundle push <REF> -f -
```

But then again we diverge from kubectl in many other places so I don't mind if we avoid the `-f` option.

Reply:
Yeah I think that makes sense especially given the `start` commands also sorta do this. I will change it.